core/amber/src/main/scala/edu/uci/ics/texera/web/resource/auth/UserResource.scala

```diff
@@ -63,6 +105,52 @@ class UserResource {
     Response.ok().build()
   }
 
+  @POST
+  @Path("/google-login")
+  def googleLogin(@Session session: HttpSession, request: GoogleUserLoginRequest): Response = {
+    // get authorization code from request
+    var code = request.authoCode
+
+    // use authorization code to get tokens
+    try {
```

Comment:
this try is redundant when there is no catch attached. we should handle the exceptions, and report accordingly to the frontend (with a suitable HTTP status code). It is not a good idea to leave exceptions uncaught and have 500 on the frontned.

Reply:
add several error messages and set the server status to 401 (UNAUTHORIZED).